### PR TITLE
Add support for ips

### PR DESCRIPTION
**Pull request by RabsonJ**
- Refactor agent to read the port and schema if present in the `Message`, else read from the args
- Add support for scanning ips

### CODE REVIEW

File: agent/whatweb_agent.py [hunk 1]
```diff
@@ -81,38 +81,97 @@ def process(self, message: msg.Message) -> None:
...
-        is_target_already_processed = self._is_target_already_processed(message)
-        if is_target_already_processed is False:
-            return
-        else:
-            with tempfile.NamedTemporaryFile() as fp:
-                self._start_scan(target.domain, fp.name)
-                self._parse_emit_result(target.domain, fp, int(target.port), target.schema)
+        targets = self._prepare_targets(message)
+        for target in targets:
+            is_target_already_processed = self._is_target_already_processed(
```
Comment (amine3) [inline agent/whatweb_agent.py:86]: no need to use the variable is_target_already_processed, you can do the if check directly on the function result

File: agent/whatweb_agent.py [hunk 2]
```diff
@@ -81,38 +81,97 @@ def process(self, message: msg.Message) -> None:
...
+        """Returns a list of domain targets to be scanned."""
+        targets = []
         if message.data.get('url') is not None:
             target = self._get_target_from_url(message.data['url'])
+            targets.append(target)
         elif message.data.get('name') is not None:
             domain_name = message.data['name']
-            target = Target(domain=domain_name, schema=self.args.get('schema'), port=self.args.get('port'))
+
+            if message.data.get('port') is not None:
```
Comment (amine3) [inline agent/whatweb_agent.py:114]: can you add a comment on this part

File: agent/whatweb_agent.py [hunk 3]
```diff
@@ -81,38 +81,97 @@ def process(self, message: msg.Message) -> None:
...
+            try:
+                addresses = ipaddress.ip_network(f'{host}/{mask}')
+                for address in addresses.hosts():
+                    targets.append(Target(name=str(address), schema=schema, port=port))
+            except ValueError as e:
+                logger.error('Invalid IP or mask. %s', e)
+        else:
+            try:
+                addresses = ipaddress.ip_network(host)
+                for address in addresses.hosts():
```
Comment (amine3) [inline agent/whatweb_agent.py:154]: no need for the for loop here since it is a single address, you can just do return str(ipaddress.ip_network(host))

**Review by amine3 — CHANGES_REQUESTED** [review]

File: agent/whatweb_agent.py [hunk 4]
```diff
@@ -81,39 +81,112 @@ def process(self, message: msg.Message) -> None:
...
+                return True
+            else:
+                logger.info('target %s/ was processed before, exiting', unicity_check_key)
+                return False
+        elif message.data.get('host') is not None:
+            host = message.data.get('host')
+            mask = message.data.get('mask')
+            if mask is not None:
+                addresses = ipaddress.ip_network(f'{host}/{mask}')
+                result = self.add_ip_network('agent_whois_ip_asset', addresses, lambda net: f'X_{net}_Y')
```
Comment (3asm) [inline agent/whatweb_agent.py:182]: The lambda must include schema and port.
lambda net : f'{protocol}_{net}_{protocol}'

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**